swerve100/src/main/java/frc/robot/autonomous/Circle.java

```diff
@@ -42,8 +43,7 @@ public void initialize() {
             m_robotDrive.thetaController,
             () -> new Rotation2d(),
             m_robotDrive::setModuleStates,
-            m_robotDrive,
-            m_robotDrive.m_gyro,
+            gyro,
```

Comment:
this value of "gyro" will always be null because you didn't assign anything to it.